src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -51,120 +57,138 @@ def build_expr_from_python_val(expr_builder: proto.Expr, obj: Any) -> None:
-                build_expr_from_python_val(expr_builder.sp_row.vs.add(), field)
+                build_expr_from_python_val(sp_row_ast.vs.add(), field, caller_frame_depth=caller_frame_depth+1)
 
     elif isinstance(obj, bool):
-        set_src_position(expr_builder.bool_val.src)
-        expr_builder.bool_val.v = obj
+        bool_val_ast = with_src_position(expr_builder.bool_val, caller_frame_depth=caller_frame_depth)
+        bool_val_ast.v = obj
 
     elif isinstance(obj, int):
-        set_src_position(expr_builder.int64_val.src)
-        expr_builder.int64_val.v = obj
+        int64_val_ast = with_src_position(expr_builder.int64_val, caller_frame_depth=caller_frame_depth)
+        int64_val_ast.v = obj
 
     elif isinstance(obj, float):
-        set_src_position(expr_builder.float64_val.src)
-        expr_builder.float64_val.v = obj
+        float64_val_ast = with_src_position(expr_builder.float64_val, caller_frame_depth=caller_frame_depth)
+        float64_val_ast.v = obj
 
     elif isinstance(obj, str):
-        set_src_position(expr_builder.string_val.src)
-        expr_builder.string_val.v = obj
+        string_val_ast = with_src_position(expr_builder.string_val, caller_frame_depth=caller_frame_depth)
+        string_val_ast.v = obj
 
     elif isinstance(obj, bytes):
-        set_src_position(expr_builder.binary_val.src)
-        expr_builder.binary_val.v = obj
+        binary_val_ast = with_src_position(expr_builder.binary_val, caller_frame_depth=caller_frame_depth)
+        binary_val_ast.v = obj
 
     elif isinstance(obj, bytearray):
-        set_src_position(expr_builder.binary_val.src)
-        expr_builder.binary_val.v = bytes(obj)
+        binary_val_ast = with_src_position(expr_builder.binary_val, caller_frame_depth=caller_frame_depth)
+        binary_val_ast.v = bytes(obj)
 
     elif isinstance(obj, decimal.Decimal):
-        set_src_position(expr_builder.big_decimal_val.src)
+        big_decimal_val_ast = with_src_position(expr_builder.big_decimal_val, caller_frame_depth=caller_frame_depth)
         dec_tuple = obj.as_tuple()
         unscaled_val = reduce(lambda val, digit: val * 10 + digit, dec_tuple.digits)
         if dec_tuple.sign != 0:
             unscaled_val *= -1
         req_bytes = (unscaled_val.bit_length() + 7) // 8
-        expr_builder.big_decimal_val.unscaled_value = unscaled_val.to_bytes(
+        big_decimal_val_ast.unscaled_value = unscaled_val.to_bytes(
+        big_decimal_val_ast.unscaled_value = unscaled_val.to_bytes(
             req_bytes, "big", signed=True
         )
-        expr_builder.big_decimal_val.scale = dec_tuple.exponent
+        big_decimal_val_ast.scale = dec_tuple.exponent
+        big_decimal_val_ast.scale = dec_tuple.exponent
 
     elif isinstance(obj, datetime.datetime):
-        set_src_position(expr_builder.python_timestamp_val.src)
+        python_timestamp_val_ast = with_src_position(expr_builder.python_timestamp_val, caller_frame_depth=caller_frame_depth)
```

Comment:
IMO these names are too long for local variables given the simple context they're in (and therefore the lack of need for any kind of disambiguation). i would just them `ast`.